src/txmempool.cpp

```diff
@@ -737,7 +738,7 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
             waitingOnDependants.push_back(&(*it));
         else {
             CValidationState state;
-            assert(CheckInputs(tx, state, mempoolDuplicate, false, 0, false, NULL));
+            assert(Consensus::CheckTxInputs(tx, state, mempoolDuplicate, nSpendHeight));
```

Comment:
I didn't know we had just a rule, is there a place where those "rules" are documented?
If we do, I suggest we change that rule to at least not apply to CValidationState.
In any case, this is not a change in this PR: there's no change in functionality here, it was doing the same before.